packages/legend-query-builder/src/components/result/QueryBuilderResultPanel.tsx

```diff
@@ -77,6 +77,60 @@ import { QueryBuilderTDSSimpleGridResult } from './tds/QueryBuilderTDSSimpleGrid
 import { getExecutedSqlFromExecutionResult } from './tds/QueryBuilderTDSResultShared.js';
 import { QueryBuilderTDSGridResult } from './tds/QueryBuilderTDSGridResult.js';
 import type { QueryBuilder_LegendApplicationPlugin_Extension } from '../../stores/QueryBuilder_LegendApplicationPlugin_Extension.js';
+import type { QueryBuilderResultState } from '../../stores/QueryBuilderResultState.js';
+
+const PERMISSION_ERRORS = ['permission denied', 'invalid user id or password'];
+
+export const QueryBuilderExecutionErrorPanel = observer(
+  (props: { resultState: QueryBuilderResultState }) => {
+    const { resultState } = props;
+    const error = resultState.executionError;
+    const errorMessage = error
+      ? resultState.queryBuilderState.applicationStore.notificationService.getErrorMessage(
+          error,
+        )
+      : '';
+    const isPermissionDeniedError = () =>
+      Boolean(
+        PERMISSION_ERRORS.find((e) => errorMessage?.toLowerCase().includes(e)),
+      );
+    const openCheckEntitlmentsEditor = (): void => {
+      resultState.queryBuilderState.checkEntitlementsState.setShowCheckEntitlementsViewer(
+        true,
+      );
+    };
+
+    return (
+      error && (
+        <>
+          {isPermissionDeniedError() && (
+            <div className="query-builder__result__permission-error">
+              <div className="query-builder__result__permission-error__header">
+                Entitlement / Authorization error - Please
+              </div>
+              <button
+                className="query-builder__result__permission-error__button"
+                onClick={openCheckEntitlmentsEditor}
+              >
+                Click Here to Check Entitlements
```

Comment:
should we only show this if entitlements view is available. is there anyway we can know if we have ability to check proper entitlements. 
Would be bad user experience if this shows up but doesn't work.

Reply:
I added the same disable condition for `Check Entitlements` here 
However, Studio will never know what engine will return without actually running APIs therefore the result panel can't predict if proper entitlements will be returned.